### DISCUSSION TITLE
string formatting, format specifiers?

### DISCUSSION THREAD

**Discussion by uvtc** [janet-lang/janet#954] [General]
Where can I find docs on the list of string format specifiers that Janet supports in `string/format` and `printf`?

**Comment by sogaiu** [comment]
@uvtc I think there is a list in [this section of the website docs](https://janet-lang.org/capi/writing-c-functions.html#Panicking) that might be helpful.

**Comment by uvtc** [comment]
Ah, thanks, @sogaiu !

I see that `(string/format "%p" x)` works nicely for many Janet data structures that I often print out with `pp`.

**Comment by sogaiu** [comment]
I tend to use `(printf "some label: %p" x)` when printing, though I also find `pp` to be convenient for unlabeled things.

On a side note, I often end up using `string/format` in `assert` forms.
